### DISCUSSION TITLE
Town Deletion from Inactivity

### DISCUSSION THREAD

**Discussion by PrismaticCardinal** [TownyAdvanced/Towny#7670] [Ideas]
I think it would be neat to have a config option where towns are deleted if no residents are active in x number of days. For example, if a town has 5 residents, and all five have been inactive for 90 days, the town will be deleted.

This could cover a wide range of use-cases, such as clearing out inactive towns, using activity-based upkeep as opposed to currency, and saving staff time sorting through the town list for the one-day-players who make claims and never come back.

**Comment by LlmDl** [comment]
Turn on the old resident deletion feature, set it to 90 days, then watch as the inactive residents are removed from the towns. When the last resident is removed from a town, that town will be deleted.

**Reply by PrismaticCardinal** [reply]
While I agree it's very similar, players have expressed a distaste for this as it directly affects active towns, kicking out members they have emotional ties to and lowering their claim chunks. Waiting until every member is inactive for 90 days would ensure only inactive towns are affected.

I of course fully understand if it's a feature you don't see worth pursuing however; it's possible that it is something not many servers would use, and I know development work takes a lot of time and effort. I sincerely appreciate that you took the time to reply to me though!

**Reply by LlmDl** [reply]
If you find that you want this for your server as a custom plugin, and you're wanting to pay to have it made, reach out to me on the Discord.
